src/main/java/de/mediathekview/mserver/crawler/orf/tasks/OrfFilmDetailTask.java

```diff
@@ -0,0 +1,227 @@
+    final Optional<LocalDateTime> aTime,
+    final Optional<Duration> aDuration) {
+    
+    try {
+      if (aVideoInfo.isPresent() && aTitle.isPresent()) {
+        final Film film = new Film(UUID.randomUUID(), crawler.getSender(), aTitle.get(),
+          aUrlDTO.getTheme(), aTime.orElse(LocalDateTime.now()), aDuration.orElse(Duration.ZERO));
+
+        film.setWebsite(new URL(aUrlDTO.getUrl()));
+        if (aDescription.isPresent()) {
+          film.setBeschreibung(aDescription.get());
+        }
+     
+        OrfVideoInfoDTO videoInfo = aVideoInfo.get();
+        if (StringUtils.isNotBlank(videoInfo.getSubtitleUrl())) {
+          film.addSubtitle(new URL(videoInfo.getSubtitleUrl()));
+        }
+        
+        addUrls(film, videoInfo.getVideoUrls());
+        film.setGeoLocations(CrawlerTool.getGeoLocations(crawler.getSender(), videoInfo.getDefaultVideoUrl()));
+
+        taskResults.add(film);
+        crawler.incrementAndGetActualCount();
+        crawler.updateProgress();
+      } else {
+        LOG.error("OrfFilmDetailTask: no title or video found for url " + aUrlDTO.getUrl());
+        crawler.incrementAndGetErrorCount();
+        crawler.updateProgress();
+      }
+    } catch (MalformedURLException ex) {
+      LOG.fatal("A ORF URL can't be parsed.", ex);
+      crawler.printErrorMessage();
+      crawler.incrementAndGetErrorCount();
+      crawler.updateProgress();
+    }      
+  }
+
+  private void addUrls(final Film aFilm, final Map<Resolution, String> aVideoUrls)
+    throws MalformedURLException {
+    
+    for (final Map.Entry<Resolution, String> qualitiesEntry : aVideoUrls.entrySet()) {
+      aFilm.addUrl(qualitiesEntry.getKey(), CrawlerTool.stringToFilmUrl(qualitiesEntry.getValue()));
+    }
+  }
+  
+  private Optional<OrfVideoInfoDTO> parseUrls(Document aDocument) {
+    Optional<String> json = HtmlDocumentUtils.getElementAttributeString(VIDEO_SELECTOR, ATTRIBUTE_DATA_JSB, aDocument);
+    
+    if (json.isPresent()) {
+    
+      final Gson gson = new GsonBuilder().registerTypeAdapter(OPTIONAL_VIDEOINFO_TYPE_TOKEN,
+          new OrfVideoDetailDeserializer()).create();
+      
+      return gson.fromJson(json.get(), OPTIONAL_VIDEOINFO_TYPE_TOKEN);
+    }
+    
+    return Optional.empty();
+  }
+  
+  private static Optional<LocalDateTime> parseDate(Document aDocument) {
+    Optional<String> date = HtmlDocumentUtils.getElementAttributeString(TIME_SELECTOR, ATTRIBUTE_DATETIME, aDocument);
+    if (date.isPresent()) {
+      String dateValue = date.get().replace("CET", " ").replace("CEST", " ");
+      try {
+        LocalDateTime localDate = LocalDateTime.parse(dateValue, DATE_TIME_FORMATTER);
+        return Optional.of(localDate);
+      } catch(DateTimeParseException e) {
+        LOG.debug("OrfFilmDetailTask: unknown date format: " + date.get());
+      }
+    }
+    
+    return Optional.empty();
+  }
+    
+  private static Optional<Duration> parseDuration(Document aDocument) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)